frontend/dockerfile/dockerfile_lint_test.go

```diff
@@ -208,6 +209,37 @@ COPY Dockerfile /bar
 	checkLinterWarnings(t, sb, dockerfile, []expectedLintWarning{})
 }
 
+func testUndeclaredArg(t *testing.T, sb integration.Sandbox) {
+	dockerfile := []byte(`
+ARG base=scratch
+FROM $base
+COPY Dockerfile .
+`)
+	checkLinterWarnings(t, sb, dockerfile, []expectedLintWarning{})
+
+	dockerfile = []byte(`
+ARG platform=linux/amd64
+FROM --platform=$platform scratch
+COPY Dockerfile .
+`)
+	checkLinterWarnings(t, sb, dockerfile, []expectedLintWarning{})
```

Comment:
Add a testcase for common case:

```
FROM --platform=$BULIDPLATFORM scratch
```

(intentional typo)

While no warning for `--platform=$BUILDPLATFORM`